internal/services/orbital/helper.go

```diff
@@ -17,13 +18,161 @@ type SpacecraftLinkModel struct {
 	Name               string  `tfschema:"name"`
 }
 
+type ContactProfileLinkModel struct {
+	Polarization        string                       `tfschema:"polarization"`
+	Direction           string                       `tfschema:"direction"`
+	Name                string                       `tfschema:"name"`
+	Channels            []ContactProfileChannelModel `tfschema:"channels"`
+	GainOverTemperature float64                      `tfschema:"gain_over_temperature"`
+	EirpDbw             float64                      `tfschema:"eirp_dbw"`
+}
+
+type ContactProfileChannelModel struct {
+	BandwidthMhz              float64         `tfschema:"bandwidth_mhz"`
+	CenterFrequencyMhz        float64         `tfschema:"center_frequency_mhz"`
+	EndPoint                  []EndPointModel `tfschema:"end_point"`
+	Name                      string          `tfschema:"name"`
+	ModulationConfiguration   string          `tfschema:"modulation_configuration"`
+	DemodulationConfiguration string          `tfschema:"demodulation_configuration"`
+}
+
 type EndPointModel struct {
 	EndPointName string `tfschema:"end_point_name"`
 	IpAddress    string `tfschema:"ip_address"`
 	Port         string `tfschema:"port"`
 	Protocol     string `tfschema:"protocol"`
 }
 
+func ContactProfileLinkSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:     pluginsdk.TypeList,
+		Required: true,
+		ForceNew: true,
+		MinItems: 1,
+		Elem: &pluginsdk.Resource{
+			Schema: map[string]*schema.Schema{
+				"channels": ChannelSchema(),
+
+				"direction": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(contactprofile.DirectionDownlink),
+						string(contactprofile.DirectionUplink),
+					}, true),
+				},
+
+				"name": {
+					Type:         pluginsdk.TypeString,
+					Required:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
+				},
+
+				"polarization": {
+					Type:     pluginsdk.TypeString,
+					Required: true,
+					ValidateFunc: validation.StringInSlice([]string{
+						string(contactprofile.PolarizationLHCP),
+						string(contactprofile.PolarizationRHCP),
+						string(contactprofile.PolarizationLinearVertical),
+						string(contactprofile.PolarizationLinearHorizontal),
+					}, false),
+				},
+
+				"gain_over_temperature": {
+					Type:     pluginsdk.TypeFloat,
+					Optional: true,
+				},
+
+				"eirp_dbw": {
+					Type:     pluginsdk.TypeFloat,
+					Optional: true,
+				},
+			},
+		},
+	}
+}
+
+func ChannelSchema() *pluginsdk.Schema {
```

Comment:
again we shouldn't be creating functions like this unless they are reused, it is far less readable/maintainable split out like this

Reply:
Yes, this split out is for future reuse purpose.